docs/data/base/components/number-input/number-input.md

```diff
@@ -0,0 +1,146 @@
+---
+product: base
+title: Unstyled React Number Input component and hook
+components: NumberInputUnstyled
+hooks: useNumberInput
+githubLabel: 'component: NumberInput'
+---
+
+# Unstyled Number Input
+
+<p class="description">The Unstyled Number Input component provides users with a field for integer values, and buttons to increment or decrement the value.</p>
+
+{{"component": "modules/components/ComponentLinkHeader.js", "design": false}}
+
+{{"component": "modules/components/ComponentPageTabs.js"}}
+
+## Introduction
+
+A number input is a UI element that accepts numeric values from the user.
+The Unstyled Number Input component is a customizable replacement for the native HTML `<input type="number" />` that solves common usability [issues](/material-ui/react-text-field/#type-quot-number-quot) of the native counterpart.
```

Comment:
I think it's worth exposing the "common usability issues" here instead of navigating the user to the Material UI docs (which they might not want to deal with at all). It also seems like a central argument as to why we're developing this as a standalone component... feels important. @samuelsycamore what do you think?

Reply:
I agree ~ Let me expand this part